Tests/PackageLoading/ConventionTests.swift

```diff
@@ -158,21 +158,42 @@ private extension FileSystem {
     }
 }
 
-/// Loads a package using PackageBuilder at the given path. If package description is provided it is used,
-/// otherwise a default package with name of directory is passed to manifest.
+/// A convenience enum for package builder tester. Since most of our convention tests
+/// only requires a Package object with the name filled in.
+enum PackageOrName {
+    case package(PackageDescription.Package)
+    case name(String)
+}
+
+extension PackageOrName: ExpressibleByStringLiteral {
+    public typealias UnicodeScalarLiteralType = StringLiteralType
+    public typealias ExtendedGraphemeClusterLiteralType = StringLiteralType
+    public init(stringLiteral value: String) {
+        self = .name(value)
+    }
+    public init(extendedGraphemeClusterLiteral value: String) {
+        self.init(stringLiteral: value)
+    }
+    public init(unicodeScalarLiteral value: String) {
+        self.init(stringLiteral: value)
+    }
+}
+
+/// Loads a package using PackageBuilder at the given path.
 ///
 /// - Parameters:
+///     - packageOrName: Package name or PackageDescription instance to use for loading this package.
 ///     - path: Directory where the package is located.
 ///     - in: FileSystem in which the package should be loaded from.
-///     - package: PackageDescription instance to use for loading this package.
 ///
 /// - Throws: ModuleError, ProductError
-private func loadPackage(_ path: AbsolutePath, in fs: FileSystem, package: PackageDescription.Package? = nil) throws -> PackageModel.Package {
+private func loadPackage(_ packageOrName: PackageOrName, path: AbsolutePath, in fs: FileSystem) throws -> PackageModel.Package {
```

Comment:
What if loadPackage took in the `PackageDescription.Package`, and there were just two `init` functions, one that takes a name and one that takes a `PackageDescription.Package`, and the former just makes a dummy package and calls the other. That avoids needing the extra type.